Structure_Engine/Modify/Transform.cs

```diff
@@ -249,6 +249,44 @@ public static FEMesh Transform(this FEMesh mesh, TransformMatrix transform, doub
         }
 
         /***************************************************/
+
+
+        [Description("Transforms the PadFoundation's edges, openings and basis by the transform matrix. Only rigid body transformations are supported.")]
+        [Input("padFoundation", "PadFoundation to transform.")]
+        [Input("transform", "Transform matrix.")]
+        [Input("tolerance", "Tolerance used in the check whether the input matrix is equivalent to the rigid body transformation.")]
+        [Output("transformed", "Modified PadFoundation with unchanged properties, but transformed edges and basis.")]
+        public static PadFoundation Transform(this PadFoundation padFoundation, TransformMatrix transform, double tolerance = Tolerance.Distance)
+        {
+            if (padFoundation == null)
+            {
+                Base.Compute.RecordError("Could not transform the PadFoundation because it was null.");
+                return null;
+            }
+
+            if (!transform.IsRigidTransformation(tolerance))
+            {
+                Base.Compute.RecordError("Transformation failed: only rigid body transformations are currently supported.");
+                return null;
+            }
+
+            PadFoundation result = padFoundation.ShallowClone();
+
+            List<Edge> edges = new List<Edge>();
+            foreach (Edge edge in result.TopOutline)
+            {
+                Edge clone = edge.ShallowClone();
+                clone.Curve = clone.Curve.ITransform(transform);
+                edges.Add(clone);
+            }
+
+            result.TopOutline = edges;
```

Comment:
What is happening to the orientation angle here? Should we orient it too? Probs easiest to look up how it is done for `Panel`